src/games/strategy/util/UrlStreams.java

```diff
@@ -0,0 +1,87 @@
+package games.strategy.util;
+
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Optional;
+import java.util.function.Function;
+
+import com.google.common.base.Throwables;
+
+import games.strategy.debug.ClientLogger;
+
+
+/**
+ * Utility class for opening input streams from URL and URI objects.
+ */
+public final class UrlStreams {
+
+  /**
+   * Opens an input stream to a given url. Returns Optional.empty() in case there is a failure.
+   * The failure message is logged to the user.
+   *
+   * @return Optional.empty() if there was a failure opening the strema, otherwise an optional
+   *         containing an input stream to the parameter uri.
+   */
+  public static Optional<InputStream> openStream(URL url) {
```

Comment:
Only two public methods, both static
The public static API is not actually tested, but it is very thin at least.

The rest of the class is protected so it can be tested. We have the public static API because it is more convenient to call: `UrlStreams.openStream(url)`, than it is to call: `new UrlStreams().newStream(url)`